File: Objective-C/Search/Controller/DefaultNavigationController.m
//
//  DefaultNavigationController.m
//  Search
//
//  Created by John Lima on 19/04/17.
//  Copyright © 2017 John Lima. All rights reserved.
//

#import "DefaultNavigationController.h"

@interface DefaultNavigationController ()

@end

@implementation DefaultNavigationController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self updateUI];
}

-(UIStatusBarStyle)preferredStatusBarStyle {
    return UIStatusBarStyleLightContent;
}

-(void)updateUI {
    self.navigationBar.barTintColor = [UIColor purpleColor];
    self.navigationBar.tintColor = [UIColor whiteColor];
    [self.navigationBar setTitleTextAttributes: @{NSFontAttributeName: [UIFont systemFontOfSize:17], NSForegroundColorAttributeName: [UIColor whiteColor]}];
}

@end

File: Swift/Search/Controller/DefaultNavigationController.swift
//
//  DefaultNavigationController.swift
//  Search
//
//  Created by John Lima on 19/04/17.
//  Copyright © 2017 John Lima. All rights reserved.
//

import UIKit

class DefaultNavigationController: UINavigationController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        updateUI()
    }
    
    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }
    
    fileprivate func updateUI() {
        navigationBar.barTintColor = .purple
        navigationBar.tintColor = .white
        navigationBar.titleTextAttributes = [NSFontAttributeName: UIFont.systemFont(ofSize: 17), NSForegroundColorAttributeName: UIColor.white]
    }
}
